### PR TITLE
Add start_epoch variable to base trainer train()

### PR DESCRIPTION
**Pull request by liamchalcroft**
As far as I can tell, there is no way to resume training currently without the trainer beginning from epoch 1. This should avoid overwrites and allow to continue correctly in schedulers (not tested)

### CODE REVIEW

**Comment by clementchadebec** [comment]
Hi @liamchalcroft,

Sorry for the late reply. I indeed think this is a useful feature that I will integrate in the near future. Nonetheless, I was thinking of a method called `resume_training_from_folder` that will take as input the path to a folder containing the checkpoints of the model, the optimizer and the scheduler. It will then reload their `state_dict` and launch resume the training as you propose.
